packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -222,43 +234,80 @@ export class EntityService {
     return newEntity;
   }
 
-  async createInitialEntities(appId: string, user: User): Promise<Entity[]> {
-    const app = this.prisma.app.update({
+  async createDefaultEntities(appId: string, user: User): Promise<void> {
+    return this.bulkCreateEntities(appId, user, DEFAULT_ENTITIES);
+  }
+
+  /**
+   * Creates multiple entities.
+   * The function supports creation of entities with reference to other entities.
+   * Use "[entityName]" notation for any property values and it will be replaced with the actual ID of the entity, whether an existing one or a new one.
+   * Reference to new entities must only be located after the declaration of the new entity (the order of the entities count)
+   */
+  async bulkCreateEntities(
+    appId: string,
+    user: User,
+    entities: BulkEntityData[]
+  ): Promise<void> {
+    const existingEntities = await this.prisma.entity.findMany({
       where: {
-        id: appId
-      },
-      data: {
-        entities: {
-          create: INITIAL_ENTITIES.map(entity => ({
-            name: entity.name,
-            displayName: entity.displayName,
-            pluralDisplayName: entity.pluralDisplayName,
-            description: entity.description,
-            lockedAt: new Date(),
-            lockedByUser: {
-              connect: {
-                id: user.id
-              }
-            },
-            entityVersions: {
-              create: {
-                commit: undefined,
-                versionNumber: CURRENT_VERSION_NUMBER,
-                name: entity.name,
-                displayName: entity.displayName,
-                pluralDisplayName: entity.pluralDisplayName,
-                description: entity.description,
-                entityFields: {
-                  create: entity.fields
-                }
+        appId: appId
+      }
+    });
+
+    //create a map of existing entities
+    const entityMap = Object.fromEntries(
+      existingEntities.map(entity => {
+        return [`[${entity.name}]`, entity.id];
```

Comment:
What is this [entityName] notation?

Reply:
The new entities can hold references to other entity using its name which is replaced with its actual ID.
It can be used for new and existing entities.
For example, if we want to reference the user entity by its ID, we can use the following code:

{ relatedEntityId: "[user]" }